gluoncv/model_zoo/fastscnn.py

```diff
@@ -0,0 +1,344 @@
+__all__ = ['FastSCNN', 'get_fastscnn']
+
+from mxnet.context import cpu
+from mxnet.gluon.block import HybridBlock
+from mxnet.gluon import nn
+import mxnet as mx
+
+
+class FastSCNN(HybridBlock):
+    r"""Fast-SCNN
+
+        Parameters
+        ----------
+        nclass : int
+            Number of categories for the training dataset.
+        aux : bool
+            Auxiliary loss.
+
+        Reference:
+            Fast-SCNN: Fast Semantic Segmentation Network - BMVC 2019
+        """
+    def __init__(self, nclass, aux=True, ctx=cpu(), pretrained_base=False,
+                 height=None, width=None, base_size=2048, crop_size=1024, **kwargs):
+        super(FastSCNN, self).__init__()
+        height = height if height is not None else crop_size
+        width = width if width is not None else crop_size
+        self.aux = aux
+        with self.name_scope():
+            self.learning_to_downsample = LearningToDownsample(32, 48, 64)
+            self.global_feature_extractor = GlobalFeatureExtractor(64, [64, 96, 128], 128, 6, [3, 3, 3],  height=height//32, width=width//32)
+            self.feature_fusion = FeatureFusionModule(64, 128, 128, height=height//8, width=width//8)
+            self.classifier = Classifer(128, nclass)
+
+            if self.aux:
+                self.auxlayer = _auxHead(in_channels=64, channels=64, nclass=nclass, **kwargs)
+                self.auxlayer.initialize(ctx=ctx)
+                self.auxlayer.collect_params().setattr('lr_mult', 10)
+
+        self._up_kwargs = {'height': height, 'width': width}
+        self.base_size = base_size
+        self.crop_size = crop_size
+
+    def hybrid_forward(self, F, x):
+        higher_res_features = self.learning_to_downsample(x)
+        x = self.global_feature_extractor(higher_res_features)
+        x = self.feature_fusion(higher_res_features, x)
+        x = self.classifier(x)
+        x = F.contrib.BilinearResize2D(x, **self._up_kwargs)
+
+        outputs = []
+        outputs.append(x)
+        if self.aux:
+            auxout = self.auxlayer(higher_res_features)
+            auxout = F.contrib.BilinearResize2D(auxout, **self._up_kwargs)
+            outputs.append(auxout)
+
+        return tuple(outputs)
+
+    def demo(self, x):
+        h, w = x.shape[2:]
+        self._up_kwargs['height'] = h
+        self._up_kwargs['width'] = w
+
+        higher_res_features = self.learning_to_downsample(x)
+        x = self.global_feature_extractor(higher_res_features)
+        x = self.feature_fusion(higher_res_features, x)
+        x = self.classifier(x)
+        import mxnet.ndarray as F
+        x = F.contrib.BilinearResize2D(x, **self._up_kwargs)
+        return x
+
+    def evaluate(self, x):
+        """evaluating network with inputs and targets"""
+        return self.forward(x)[0]
+
+
+class FeatureFusionModule(HybridBlock):
+    def __init__(self, highter_in_channels, lower_in_channels, out_channels, height, width, scale_factor=4, **kwargs):
```

Comment:
You need `norm_layer=nn.BatchNorm, norm_kwargs=None` here in the '__inti__' function, so that you can use syncbn afterwards. Change line 88 to `norm_layer` later.